### PR TITLE
Feature/cloud 903 variant handler charts

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Variant-handler charts

Fixes [#34](https://usxtech.atlassian.net/browse/CLOUD-903)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create sample values.yaml file
  - Example:  I am using below niginx image to deploy
  - 
        ```yaml
        # This file contains the defaults for all optional, overridable values
        
        # This file contains the defaults for all optional, overridable values
        
        serviceMonitor:
          interval: 10s
          scrapeTimeout: 10s
        
        service:
          port: 80
          targetPort: 80
          healthCheckPort:
          metricsPort:
          healthCheckPath: /
        
        serviceAccount:
          roleArn:
        
        replicaCount: 1
        
        autoscaling:
          enabled: false
          minReplicas: 1
          maxReplicas: 5
          targetCPUUtilizationPercentage: 80
          targetMemoryUtilizationPercentage:
        
        deployment:
          image:
            tag: k8s.gcr.io/nginx-slim:0.8
            pullPolicy: IfNotPresent
          resources:
            limits:
              cpu: 1
              memory: 768Mi
            requests:
              cpu: .1
              memory: 384Mi
          args: []
          podAnnotations: {}
        
        revision: abc
        
        awsSecrets:
          - name: test-key
        
        secretVars:
          foo: bar
        
        configVars:
          bar: foo
        
        nodeSelector: {}
        tolerations: []
        
        affinity: {}
        podSecurityContext:
            fsGroup: 65534
        
        securityContext: {}
        
        # secrets1:
        #   nginxConf: hello
        ```  
Test Case 1 [Deploy nginx with new variant handler charts]:
- Steps
  - cd to variant handler folder and run 'helm template .' to check the template
  - Create namespace or use any existing one
  - To dry-run 'helm install nginx --namespace naveen --dry-run . -f ../../temp/handler-test.yaml'
  - To deploy helm install nginx --namespace naveen . -f ../../temp/handler-test.yaml
- Result
 - If everything looks good and nginx is running 
 - Uninstall the app using ' helm uninstall nginx -n naveen'
 - Result: Passed [Ralph]
```

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: charts/variant-handler/values.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+secrets: []
+  # - name: eng-secret-in-aws
+  #   mountPath: /app/secrets
+  #   fileName: aws-secret.json
+
+secretsEnvVars: []
+  # foo: bar
+
+configsEnvVars: []
+  # bar: foo
```
Comment (rchengvariant) [inline charts/variant-handler/values.yaml:63]: Nodeslector
Tolerations
Affinity

File: charts/variant-handler/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        fsGroup: 65534
+      {{- if len $secrets }}
+      volumes:
+        {{- range $secrets }}
+        - name: {{ .name }}
+          secret:
+            secretName: {{ $fullName }}-{{ .name }}
+        {{- end }}
+      {{- end }}
+      containers:
```
Comment (LeeHughsVariant) [inline charts/variant-handler/templates/deployment.yaml:75]: Missing the rest of the security context info.
runAsNonRoot: true
capabilities:
   drop:
   - ALL
allowPrivilegeEscalation: false
readOnlyRootFilesystem: true

Comment (naveen-variant) [reply charts/variant-handler/templates/deployment.yaml:75]: Where are we using in the deployment file.

**Comment by rchengvariant** [comment]
```yaml
Test Case 1 [Deploy nginx with new variant handler charts]:
- Steps
  - created sample.yaml with provided yaml 
  - cd to variant handler folder and run 'helm template .' to check the template
  - Create namespace or use any existing one (naveen)
  - To dry-run 'helm install nginx-test --namespace naveen --dry-run . -f ../../temp/handler-test.yaml'
  - To deploy helm install nginx-test --namespace naveen . -f ../../temp/handler-test.yaml
      coalesce.go:175: warning: skipped value for variant-handler.secretVars: Not a table.
      coalesce.go:175: warning: skipped value for variant-handler.configVars: Not a table.
      NAME: nginx-test
      LAST DEPLOYED: Mon Feb 14 11:40:19 2022
      NAMESPACE: naveen
      STATUS: deployed
      REVISION: 1
      TEST SUITE: None
- Result
 - If everything looks good and nginx is running 
 - Uninstall the app using ' helm uninstall nginx-test -n naveen'
 - Result: PASSED [Ralph]
 ```

**Review by hpatel826 — APPROVED** [review]

**Pull request merged**